java/ql/src/Security/CWE/CWE-940/AndroidIntentRedirectionSample.java

```diff
@@ -0,0 +1,18 @@
+// BAD: A user-provided Intent is used to launch an arbitrary component
+Intent forwardIntent = (Intent) getIntent().getParcelableExtra("forward_intent");
+startActivity(forwardIntent);
+
+// GOOD: The destination component is checked before launching it
```

Comment:
Does the query as written implement this sort of sanitization?

Reply:
It didn't. I wasn't sure if adding these checks as sanitizers/guards would be accurate enough, but since you mentioned it I went ahead and gave it a shot in 22038b5. Let me know what you think.